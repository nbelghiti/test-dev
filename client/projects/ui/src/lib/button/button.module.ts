import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { ButtonComponent } from './button/button.component';
import { IconButtonComponent } from './icon-button/icon-button.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
	],
	declarations: [
		ButtonComponent,
		IconButtonComponent,
	],
	exports: [
		ButtonComponent,
		IconButtonComponent,
	],
})
export class ButtonModule {}
