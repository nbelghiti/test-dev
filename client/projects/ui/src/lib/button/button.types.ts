export enum ButtonType {
	primary = '',
	secondary = 'c-button--secondary',
	tertiary = 'c-button--tertiary',
	destructive = 'c-button--destructive',
}
