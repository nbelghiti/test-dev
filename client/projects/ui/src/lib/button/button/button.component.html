<button
	*ngIf="!link; else isLink"
	class="c-button {{ type }} {{ classes }}"
	[title]="buttonTitle"
	[ngClass]="{
		'c-button--disabled': disabled,
		'c-button--loading': loading,
		'c-button--fluid': fluid
	}"
	[disabled]="disabled"
	(click)="loading ? null : clicked.emit($event)"
>
	<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
</button>

<ng-template #isLink>
	<a
		class="c-button {{ type }} {{ classes }}"
		[ngClass]="{
			'c-button--disabled': disabled,
			'c-button--loading': loading,
			'c-button--fluid': fluid
		}"
		[routerLink]="!disabled && link"
	>
		<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
	</a>
</ng-template>

<ng-template #buttonContent>
	<div
		*ngIf="iconBefore"
		class="c-button__icon c-button__icon--before"
	>
		<ui-icon
			[name]="iconBefore"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>

	<span class="c-button__text">
		<ng-content></ng-content>
	</span>

	<i class="c-button__loader fas fa-spinner fa-spin"></i>

	<div
		*ngIf="iconAfter"
		class="c-button__icon c-button__icon--after"
	>
		<ui-icon
			[name]="iconAfter"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>
</ng-template>
