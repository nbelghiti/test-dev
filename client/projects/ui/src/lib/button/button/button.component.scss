// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Button
// =============================================================================
.c-button {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;
	min-width: 8rem;

	padding: 1rem 2rem;

	text-align: center;
	text-decoration: none;

	border: $border-width-base solid $primary-color;
	background-color: $primary-color;
	color: $text-color-white;

	transition: background-color $animation-base, border-color $animation-base;

	cursor: pointer;

	&:hover {
		background-color: transparent;
		color: $primary-color;
	}

	&:active {
		background-color: rgba($primary-color, 0.08);

		transition: background-color 0s;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-button__icon {
	display: flex;

	font-size: 1.5rem;
	line-height: $line-height-base;

	color: currentColor;

	transition: color $animation-base;
}

.c-button__text {
	display: inline-block;
	max-width: 25rem;
	overflow: hidden;

	font-family: $font-family-secondary;
	font-weight: $font-weight-bold;
	font-size: $font-size-base;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-overflow: ellipsis;
	white-space: nowrap;

	color: currentColor;

	transition: color $animation-base;
}

.c-button__loader {
	position: absolute;

	display: none;

	transform: translate(-50%, -50%);
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-button--secondary,
.c-button--tertiary {
	border: $border-width-base solid $border-color-grey-76;
	background-color: $background-color-base;
	color: $text-color-grey-46;

	&:active {
		background-color: rgba($background-color-black, 0.08);

		transition: background-color $animation-base;
	}

	&:focus {
		outline-color: $border-color-black;
	}

	&:hover {
		border-color: $border-color-black;
		color: $text-color-base;
	}
}

.c-button--tertiary {
	padding: 0.8rem 1rem;

	.c-button__icon {
		line-height: $line-height-normalize;
	}

	.c-button__text {
		font-size: $font-size-2x-small;
	}

	.c-button__icon--after {
		margin-left: 0.8rem;
	}

	.c-button__icon--before {
		margin-right: 0.8rem;
	}
}

.c-button__icon--after {
	margin-left: 1rem;
}

.c-button__icon--before {
	margin-right: 1rem;
}

.c-button--disabled:not(.c-button--secondary):not(.c-button--tertiary) {
	background-color: $background-color-grey-86;
	border-color: $background-color-grey-86;

	pointer-events: none;
}

.c-button--secondary,
.c-button--tertiary {

	&.c-button--disabled {
		color: $text-color-grey-76;

		pointer-events: none;
	}
}

.c-button--loading {
	pointer-events: none;

	.c-button__text,
	.c-button__icon {
		opacity: 0;
	}

	.c-button__loader {
		display: block;
	}
}

.c-button--fluid {
	width: 100%;

	.c-button__text {
		max-width: 100%;
	}
}

.c-button--destructive {
	color: $ui-color-error-dark;
	background-color: $ui-color-error-light;
	border-color: $ui-color-error-light;

	&:hover,
	&:focus,
	&:active {
		color: $ui-color-error-light;
		background-color: $ui-color-error-dark;
		border-color: $ui-color-error-dark;
	}
}
