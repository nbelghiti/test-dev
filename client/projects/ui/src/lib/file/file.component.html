<div
	class="c-file"
>
	<div class="c-file__left" [title]="label">
		<div *ngIf="iconBefore" class="c-file__icon">
			<ui-icon [name]="iconBefore"></ui-icon>
		</div>

		<span class="c-file__label">{{label}}</span>

		<ng-content></ng-content>
	</div>

	<div *ngIf="url" class="c-file__right">
		<span class="c-file__last-updates">
			{{ 'FILE__LAST_GENERATED' | translate }}: {{lastGenerated}}
		</span>

		<div class="c-file__link">
			<ui-button
				[type]="ButtonType.tertiary"
				[iconAfter]="IconName.externalLink"
				[link]="url"
			>
				{{ 'FILE__VAULT_LINK_lABEL' | translate }}
			</ui-button>
		</div>
	</div>
</div>
