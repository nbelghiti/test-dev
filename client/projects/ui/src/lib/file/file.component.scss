// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: File
// =============================================================================
.c-file {
	display: inline-flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	height: 8rem;
	padding-right: 2.5rem;
	padding-left: 1.6rem;

	text-decoration: none;

	background-color: $background-color-base;

	color: $text-color-base;

	@include respond-at(get-breakpoint("desktop")) {
		height: 4.4rem;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-file__label {
	max-width: 32rem;
	overflow: hidden;
	margin-right: 1rem;

	font-size: $font-size-1x-small;
	text-overflow: ellipsis;
	white-space: nowrap;
}

.c-file__left,
.c-file__right {
	display: inline-flex;
	align-items: center;
}

.c-file__last-updates {
	font-size: $font-size-1x-small;
	white-space: nowrap;

	color: $text-color-grey-46;
}

.c-file__link {
	margin-left: 1rem;
}

.c-file__right {
	flex-wrap: wrap;
	justify-content: flex-end;
	margin-left: 2rem;
}

.c-file__icon {
	margin-right: 1.6rem;
}
