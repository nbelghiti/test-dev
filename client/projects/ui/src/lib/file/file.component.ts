import { Component, HostBinding, Input } from '@angular/core';

import { ButtonType } from '../button/button.types';
import { IconName } from '../icon/icon.types';

@Component({
	selector: 'ui-file',
	templateUrl: './file.component.html',
	styleUrls: ['./file.component.scss']
})

export class FileComponent {
	@HostBinding('class.u-full-width') public fullWidthClassName = true;

	@Input() public label: string;
	@Input() public url: string;
	@Input() public lastGenerated: string = new Date().getDay().toString();
	@Input() public iconBefore: string;

	public ButtonType: typeof ButtonType = ButtonType;
	public IconName: typeof IconName = IconName;
}
