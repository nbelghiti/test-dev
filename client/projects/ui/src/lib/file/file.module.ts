import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonModule } from '../button/button.module';
import { IconModule } from '../icon/icon.module';

import { FileComponent } from './file.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
		ButtonModule,
		TranslateModule
	],
	declarations: [
		FileComponent,
	],
	exports: [
		FileComponent,
	],
})
export class FileModule {}
