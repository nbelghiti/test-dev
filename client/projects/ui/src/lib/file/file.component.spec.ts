import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonComponent } from '../button/button/button.component';
import { IconComponent } from '../icon/icon.component';

import { FileComponent } from './file.component';

describe('FileComponent', () => {
	let component: FileComponent;
	let fixture: ComponentFixture<FileComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				TranslateModule.forRoot(),
			],
			declarations: [
				FileComponent,
				IconComponent,
				ButtonComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(FileComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
