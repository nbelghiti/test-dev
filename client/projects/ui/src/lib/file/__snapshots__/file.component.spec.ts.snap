// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileComponent Should render correctly 1`] = `
<ui-file
  ButtonType={[Function Object]}
  IconName={[Function Object]}
  fullWidthClassName={[Function Boolean]}
  lastGenerated={[Function String]}
>
  <div
    class="c-file"
  >
    <div
      class="c-file__left"
      title="undefined"
    >
      
      <span
        class="c-file__label"
      >
        
      </span>
    </div>
    
  </div>
</ui-file>
`;
